test/ICSharpCode.SharpZipLib.Tests/Tar/TarTests.cs

```diff
@@ -899,5 +899,77 @@ public void StreamWithJapaneseName(int length, string encodingName)
 				File.WriteAllBytes(Path.Combine(Path.GetTempPath(), $"jpnametest_{length}_{encodingName}.tar"), memoryStream.ToArray());
 			}
 		}
+		/// <summary>
+		/// This test could be considered integration test. it creates a tar archive with the root directory specified
+		/// Then extracts it and compares the two folders. This used to fail on unix due to issues with root folder handling
+		/// in the tar archive.
+		/// </summary>
+		[Test]
+		[Category("Tar")]
+		public void rootPathIsRespected()
+		{
+			// create dummy folder structure
+			var tempDirectory = Path.Combine(Path.GetTempPath(), "sharpziplib_tar_test_folder");
+			CreateAndClearDirectory(tempDirectory);
+			using (var dummyfile = File.Create(Path.Combine(tempDirectory, "dummyfile"))) 
+			{
+				using (var randomStream = new ChaosStream())
+				{
+					randomStream.CopyTo(dummyfile);
+				}
+			}
+
+			var tarFileName = Path.Combine(Path.GetTempPath(), "sharpziplib_tar_test_folder_archive.tar");
+
+			using (var tarFile = File.Open(tarFileName, FileMode.Create))
+			{
+				using (var tarOutputStream = TarArchive.CreateOutputTarArchive(tarFile))
+				{
+					tarOutputStream.RootPath = tempDirectory;
+					var entry = TarEntry.CreateEntryFromFile(tempDirectory);
+					tarOutputStream.WriteEntry(entry, true);
+				}
+			}
+
+			var extractDirectory = Path.Combine(Path.GetTempPath(), "sharpziplib_tar_extract_folder");
+			CreateAndClearDirectory(extractDirectory);
+			using (var file = File.OpenRead(tarFileName))
+			{
+				using (var archive = TarArchive.CreateInputTarArchive(file, Encoding.UTF8))
+				{
+					archive.ExtractContents(extractDirectory);
+				}
+			}
+
+			var expectationDirectory = new DirectoryInfo(tempDirectory);
+			foreach (var checkFile in expectationDirectory.GetFiles("", SearchOption.AllDirectories))
+			{
+				var relativePath = checkFile.FullName.Substring(expectationDirectory.FullName.Length + 1);
+				FileAssert.Exists(Path.Combine(extractDirectory, relativePath));
+				FileAssert.AreEqual(checkFile.FullName, Path.Combine(extractDirectory, relativePath));
+			}
+		}
+
+		private void CreateAndClearDirectory(string path)
+		{
+			if (Directory.Exists(path))
+			{
+				Directory.Delete(path, true);
+			}
+			Directory.CreateDirectory(path);
+		}
+
+		public class ChaosStream : MemoryStream
+		{
+			private readonly int length = new Random().Next() % 500000 + 200;
+
+			// Create constructors as needed to match desired MemoryStream construction
+
+			public override int Read(byte[] buffer, int offset, int count)
+			{
+				int readCount = Math.Max(0, Math.Min(length - offset, count));
+				return base.Read(buffer, offset, readCount);
+			}
+		}
```

Comment:
test/ICSharpCode.SharpZipLib.Tests/TestSupport/Utils.cs contains helpers that does this already (and cleans up after tests, regardless of outcome)
```suggestion
```